constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/MotionCarousel.kt

```diff
@@ -0,0 +1,396 @@
+    numSlots: Int,
+    startState: String = "start",
+    backwardState: String = "backward",
+    forwardState: String = "forward",
+    slotPrefix: String = "card",
+    showSlots: Boolean = false,
+    content: MotionCarouselScope.() -> Unit
+) {
+
+    val SwipeStateStart = startState
+    val SwipeStateForward = forwardState
+    val SwipeStateBackward = backwardState
+
+    val provider = rememberStateOfItemsProvider(content)
+
+    var componentWidth by remember { mutableStateOf(1000f) }
+    val swipeableState = rememberSwipeableState(SwipeStateStart)
+    var mprogress = (swipeableState.offset.value / componentWidth)
+
+    var state by remember {
+        mutableStateOf(
+            CarouselState(
+                MotionCarouselDirection.FORWARD,
+                0,
+                0,
+                false,
+                false
+            )
+        )
+    }
+    var currentIndex = remember { mutableStateOf(0) }
+
+    val anchors = if (currentIndex.value == 0) {
+        mapOf(0f to SwipeStateStart, componentWidth to SwipeStateForward)
+    } else if (currentIndex.value == provider.value.count() - 1) {
+        mapOf(-componentWidth to SwipeStateBackward, 0f to SwipeStateStart)
+    } else {
+        mapOf(
+            -componentWidth to SwipeStateBackward,
+            0f to SwipeStateStart,
+            componentWidth to SwipeStateForward
+        )
+    }
+
+    val transitionName = remember {
+        mutableStateOf(forwardState)
+    }
+
+    if (mprogress < 0 && state.index > 0) {
+        state.direction = MotionCarouselDirection.BACKWARD
+        transitionName.value = backwardState
+        mprogress *= -1
+    } else {
+        state.direction = MotionCarouselDirection.FORWARD
+        transitionName.value = forwardState
+    }
+
+    if (!swipeableState.isAnimationRunning) {
+        if (state.direction == MotionCarouselDirection.FORWARD
+            && swipeableState.currentValue.equals(SwipeStateForward)
+        ) {
+            LaunchedEffect(true) {
+                if (state.index + 1 < provider.value.count()) {
+                    state.index++
+                    swipeableState.snapTo(SwipeStateStart)
+                    state.direction = MotionCarouselDirection.FORWARD
+                }
+            }
+        } else if (state.direction == MotionCarouselDirection.BACKWARD
+            && swipeableState.currentValue.equals(SwipeStateBackward)
+        ) {
+            LaunchedEffect(true) {
+                if (state.index > 0) {
+                    state.index--
+                }
+                swipeableState.snapTo(SwipeStateStart)
+                state.direction = MotionCarouselDirection.FORWARD
+            }
+        }
+        currentIndex.value = state.index
+    }
+
+    MotionLayout(motionScene = motionScene,
+        transitionName = transitionName.value,
+        progress = mprogress,
+        measureFlags = MotionLayoutMeasureFlags.FULL_MEASURE, // TODO: only apply as needed
```

Comment:
This should just be 
motionFlags  = MotionFlags.LAYOUT_MEASURE_ALWAYS 
measureFlags = MotionLayoutMeasureFlags   is way to specific